src/pubsub/mod_pubsub_db_rdbms_sql.erl

```diff
@@ -263,16 +267,195 @@ update_subscription(Nidx, LU, LS, LR, SubInt, SubId) ->
 
 % ------------------- Items --------------------------------
 
--spec insert_item(Nidx :: mod_pubsub:nodeIdx(),
-                  LU :: jid:luser(),
-                  LS :: jid:lserver(),
-                  ItemId :: mod_pubsub:itemId()) -> iolist().
-insert_item(Nidx, LU, LS, ItemId) ->
-    ["INSERT INTO pubsub_items (nidx, luser, lserver, itemid)"
-     " VALUES (", esc_int(Nidx), ", ",
-     esc_string(LU), ", ",
-     esc_string(LS), ", ",
-     esc_string(ItemId), ")"].
+-spec get_items(Nidx :: mod_pubsub:nodeIdx(), gen_pubsub_node:get_item_options()) ->
+    iolist().
+get_items(Nidx, Opts) ->
+    ["SELECT ", item_columns(), " ",
+     "FROM pubsub_items "
+     "WHERE nidx=", esc_int(Nidx),
+     maybe_item_ids_filter(maps:get(item_ids, Opts, undefined)),
```

Comment:
I'm fairly sure that `rsm` parameter is passed to this function as well but I can't see where it is used in the code?

Reply:
Yeah... It was never used (not even for the mnesia backend) and we don't have tests for RSM currently... Adding tests and making sure it works is work for a separate PR, in my opinion.

Reply:
That's fine but maybe for now we could have a note in e.g. `node_flat` or `mod_pubsub_db` that RSM is not supported yet?

Reply:
I added the TODO comment in `node_flat`.